src/nav/AppNavigator.js

```diff
@@ -59,7 +59,14 @@ export const createAppNavigator = (args: {|
       'account-details': { screen: AccountDetailsScreen },
       'group-details': { screen: GroupDetailsScreen },
       auth: { screen: AuthScreen },
-      chat: { screen: ChatScreen },
+      chat: { screen: ChatScreen,
+      navigationOptions:{
```

Comment:
How did you arrive at the choice to configure the particular routes `chat` and `search`? 🙂 Did you find a reason not to apply the changes to all routes on `AppNavigator`, e.g., with [`defaultNavigationOptions`](https://reactnavigation.org/docs/4.x/stack-navigator/#stacknavigatorconfig)?

I think this is a place where consistency would be helpful; a user might get confused if they find they can sometimes use the gesture and sometimes not, in similar situations. For that reason, I think using `defaultNavigationOptions` would be a good idea.

The doc I linked to in #4036 [says](https://developer.android.com/training/gestures/gesturenav#conflicting-gestures) that we should be careful to avoid situations where two things are set up to respond to the same gesture, with one of them being this new back-navigation handling. Did you find a situation like that? If so, it might be helpful to "opt out" of the back-nav gesture for that particular screen (if we can't easily just get rid of the pre-existing gesture handling) by, e.g., setting `gestureEnabled: false` on that screen—rather than "opting in" to the feature for just one or two screens.

Reply:
> How did you arrive at the choice to configure the particular routes chat and search?

I considered the issue #4036 is being refer to giving user an alternative and a familiar way to navigate back from the message(chat) screen (swipping left to right). 

I considered to apply the same thing on the search route to make goback action( by swipping left to right ) consistent for all the options( all message, starred, mention, search) in upperTabNavigator.

>I think using defaultNavigationOptions would be a good idea.

Yes, I agree to this point.
Should I consider applying it as ```defaultNavigationOptions``` ?

>The doc I linked to in #4036 says that we should be careful to avoid situations where two things are set up to respond to the same gesture, with one of them being this new back-navigation handling. Did you find a situation like that? 

No, I could not find any situation like that. On Android no gestures are registered for the action swipe from left to right.


Also I have read in [docs](https://reactnavigation.org/docs/4.x/stack-navigator/#gestureenabled) that states gestures are enabled by default in IOS and not on Android, so a platform specific code can be written to use gestures in ```defaultNavigationOptions```.

Can this be considered as a point of consistency for users on different platforms(IOS and Android) ?.

Reply:
> Should I consider applying it as `defaultNavigationOptions` ?

Yes, let's try that.

> Also I have read in [docs](https://reactnavigation.org/docs/4.x/stack-navigator/#gestureenabled) that states gestures are enabled by default in IOS and not on Android, so a platform specific code can be written to use gestures in `defaultNavigationOptions`.

Yes, let's use [`Platform.select`](https://reactnative.dev/docs/platform-specific-code#platform-module) to apply the new settings to Android only. The new settings may coincide with the default settings on iOS, but navigation is one of the areas where iOS and Android have important design differences, so I think it makes sense for our code to treat the two platforms separately here.